src/main/java/bernie/parser/Parser.java

```diff
@@ -0,0 +1,221 @@
+        } else if (isType(Type.DELETE, input)) {
+            return Type.DELETE;
+        } else if (isType(Type.FIND, input)) {
+            return Type.FIND;
+        } else {
+            return Type.ADD;
+        }
+    }
+
+    /**
+     * Verifies the user input is of which task type
+     * @param taskType Type
+     * @param input String
+     * @return boolean to affirm if the input is of this task
+     */
+    public boolean isType(Type taskType, String input) {
+        if (taskType.equals(Type.MARK)) {
+            return input.indexOf("mark") == 0 || input.indexOf("unmark") == 0;
+        } else if (taskType.equals(Type.EMPTY)) {
+            return input.equals("");
+        }
+        return input.indexOf(taskType.name().toLowerCase()) == 0;
+    }
+
+    /**
+     * Splits a user input into an array containing parameters for handling actions accordingly,
+     * depending on taskType.
+     * @param taskType Type
+     * @param input String, a user input to be split into an array of parameters
+     * @return String[] parsedArr.
+     * For "todo": an array of 1, containing description.
+     * For "deadline": an array of 2: [description, by]
+     * For "event": an array of 2: [description, at]
+     * For "mark": an array of 2: [action, taskNum], action is either mark/unmark
+     * For "delete": an array of 1: [taskNum]
+     * For "find": an array of 1: [description]
+     */
+    public String[] getParams(Type taskType, String input) throws InvalidArgumentException {
+        String description;
+        String[] parsedArr = null;
+        switch (taskType) {
+        case TODO:
+            parsedArr = input.split("todo ");
+            description = getDescription(parsedArr, Type.TODO);
+            parsedArr = new String[]{ description };
+            break;
+        case DEADLINE:
+            parsedArr = input.split(" /by ");
+            description = getDescription(parsedArr, Type.DEADLINE);
+            String by = getTime(parsedArr, Type.DEADLINE);
+            parsedArr = new String[]{ description, by };
+            break;
+        case EVENT:
+            parsedArr = input.split(" /at ");
+            description = getDescription(parsedArr, Type.EVENT);
+            String at = getTime(parsedArr, Type.EVENT);
+            parsedArr = new String[]{ description, at };
+            break;
+        case MARK:
+            parsedArr = input.split(" ");
+            // check valid
+            checkMarkOrDeleteInput(parsedArr, Type.MARK);
+            checkValidMarkAction(parsedArr);
+            break;
+        case DELETE:
+            parsedArr = input.split(" ");
+            checkMarkOrDeleteInput(parsedArr, Type.DELETE);
+            parsedArr = new String[]{ parsedArr[1] };
+            break;
+        case FIND:
+            parsedArr = input.split("find ");
+            description = getDescription(parsedArr, Type.FIND);
+            parsedArr = new String[]{ description };
+            break;
```

Comment:
Not sure if it's possible, but maybe abstracting out the splitting of string might improve readability!